internal/pkg/agent/cmd/install.go

```diff
@@ -125,6 +134,9 @@ func installCmd(streams *cli.IOStreams, cmd *cobra.Command, args []string) error
 
 	if enroll && fleetServer == "" {
 		if url == "" {
+			if nonInteractive {
+				return fmt.Errorf("missing URL")
```

Comment:
Being that you entered non-interactive mode it would be better to use the command-line argument that is missing, as in non-interactive mode that is the only real way to provide it.

```suggestion
				return fmt.Errorf("missing required --url argument")
```